api/endpoints/off_topic_channel_names/endpoints.py

```diff
@@ -0,0 +1,142 @@
+from typing import Optional, Union
+
+from fastapi import APIRouter, Depends
+from fastapi.responses import JSONResponse
+from sqlalchemy.orm import Query, Session
+
+from api.core.database.models.api.bot import OffTopicChannelName
+from api.core.schemas import ErrorMessage
+from api.endpoints.dependencies.database import create_database_session
+
+otn = APIRouter(prefix="/off-topic-channel-names")
+
+
+def get_all_otn(db_session: Session) -> Query:
+    """Get a partial query object with .all()."""
+    return db_session.query(OffTopicChannelName).all()
+
+
+@otn.get(
+    "/",
+    status_code=200,
+    response_model=list[str],
+    responses={404: {"model": ErrorMessage}},
+)
+def get_off_topic_channel_names(
+    random_items: Optional[int] = None,
+    db_session: Session = Depends(create_database_session),
+) -> Union[JSONResponse, list[str]]:
+    """
+    ### GET /bot/off-topic-channel-names.
+
+    Return all known off-topic channel names from the database.
+
+    If the `random_items` query parameter is given, for example using...
+        $ curl 127.0.0.1:8000/api/bot/off-topic-channel-names?random_items=5
+    ... then the API will return `5` random items from the database
+    that is not used in current rotation.
+
+    When running out of names, API will mark all names to not used and start new rotation.
+
+    #### Response format
+    Return a list of off-topic-channel names:
+    >>> [
+    ...     "lemons-lemonade-stand",
+    ...     "bbq-with-bisk"
+    ... ]
+
+    #### Status codes
+    - 200: returned on success
+    - 400: returned when `random_items` is not a positive integer
+
+    ## Authentication
+    Requires a API token.
+    """
+    if not random_items:
+        queryset = get_all_otn(db_session)
+        return [offtopic_name.name for offtopic_name in queryset]
```

Comment:
That is not necessary, `all()` returns a list, and if you use a ResponseModel with the correct field name it should automatically do this for you.